packages/flutter/lib/src/material/data_table.dart

```diff
@@ -331,6 +331,7 @@ class DataTable extends StatelessWidget {
     this.horizontalMargin = 24.0,
     this.columnSpacing = 56.0,
     this.showCheckboxColumn = true,
+    this.borderThickness = 1.0,
```

Comment:
I'd prefer to call this property "dividerThickness". Borders typically encircle a widget, dividers fit in between adjacent vertical or horizontal widget edges.